OpenRA.Game/Graphics/SpriteRenderable.cs

```diff
@@ -76,7 +76,12 @@ public IModifyableRenderable WithTint(in float3 newTint, TintModifiers newTintMo
 
 		float3 ScreenPosition(WorldRenderer wr)
 		{
-			return wr.Screen3DPxPosition(pos) + wr.ScreenPxOffset(offset) - 0.5f * scale * sprite.Size;
+			var spriteShift = 0.5f * scale * sprite.Size;
+
+			// Sprites shifting by half pixels or odd sizes will cause offset mismatches
+			spriteShift = new float3((int)spriteShift.X, (int)spriteShift.Y, spriteShift.Y);
```

Comment:
* Throwing away the Z component and duplicating Y completely breaks ts!
* Nowhere else in the code uses "Shift" to refer to a sprite offset in variable names.
* Reassigning the value over itself is also something we try to avoid as a general style rule.

I suggest
```csharp
var s = 0.5f * scale * sprite.Size;
return wr.Screen3DPxPosition(pos) + wr.ScreenPxOffset(offset) - new float3((int)s.X, (int)s.Y, s.Z);
```